invisible_cities/cities/detsim_source.py

```diff
@@ -0,0 +1,24 @@
+import numpy  as np
+import tables as tb
+import pandas as pd
+
+from typing import List, Generator
+
+from invisible_cities.io.mcinfo_io import read_mchits_df
+
+#######################################
+############### SOURCE ################
+#######################################
+def load_MC(files_in : List[str]) -> Generator:
+    for filename in files_in:
+        with tb.open_file(filename) as h5in:
+            extents = pd.read_hdf(filename, 'MC/extents')
```

Comment:
This line looks misplaced. It should either go outside the `with` statement or load the data using the already-opened file rather than the filename.

Reply:
I think the same function accepts the file object, so it can be kept there changing `filename` by `h5in`.

Reply:
In that case, align with the rest

Reply:
`extents` must be a `pd.DataFrame`, personally I prefer it to be inside the with statement. But a `to_df=pd.DataFrame.from_records` should be added in the module.
What is your preference?

Reply:
Well, since you are loading everything at once you can simply use `load_mchits_df` which gives you `hits_df` directly. I had not seen this function before, but it is doing exactly what you are doing, so better use i). Otherwise, you would need to move it outside or do `extents = pd.DataFrame.from_records(h5in.root.MC.extents.read())`